File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.py
class Solution:
    def reversePrefix(self, word: str, ch: str) -> str:
        word_list = list(word)
        for i in range(len(word_list)):
            if word_list[i] == ch:
                reversed_portion = word_list[:i + 1][::-1]
                word_list[:i + 1] = reversed_portion
                break
        return "".join(word_list)

File: 2128-reverse-prefix-of-word/reverse-prefix-of-word.cpp
class Solution {
public:
    string reversePrefix(string word, char ch) {
        for (int i = 0, j = 0; j < word.size(); ++j) {
            if (word[j] == ch) {
                int L = i;
                int R = j;
                while (L < R) {
                    char tmp = word[L];
                    word[L] = word[R];
                    word[R] = tmp;
                    ++L;
                    --R;
                }
                break;
            }
        }
        return word;
    }
};